### PR TITLE
refactor/preload

### PR DESCRIPTION
**Pull request by VioletteDrch**
A bit of refactoring + ensure all assets are loaded before creating game elements (images wouldn't show up when launching the game for the first time)

### CODE REVIEW

File: frontend/scenes/GameScene.js [hunk 1]
```diff
@@ -14,68 +14,58 @@ export default class GameScene extends Phaser.Scene {
...
 
   preload() {
-    this.fetchImages();
     retrieveSoundEffects().forEach((sound) => {
       this.load.audio(sound.name, sound.url);
     });
     this.load.image("bg", "/assets/background.jpg");
+    this.fetchImages()
+        .then(() => {
+          this.load.start();
```
Comment (andrevigneaux) [inline frontend/scenes/GameScene.js:23]: why is this needed besides all the load.image(...)?

**Review by andrevigneaux — APPROVED** [review]

File: frontend/scenes/GameScene.js (hunk 1 above)
Comment (VioletteDrch) [reply frontend/scenes/GameScene.js:23]: load.image / audio etc. don't actually load the objects, it puts them in a load queue (preload). load.start() actually loads them. Without load.start, I believe the queue starts to be loaded when the create function is loaded, but then it means not every item might be loaded when the cards etc are created. So you need to ensure load.once("complete", fn) to make sure that every item in the queue has been loaded. 
It is not fully intuitive but the load complete doesn't work without the load.start.
